src/ir/possible-contents.cpp

```diff
@@ -1338,11 +1333,19 @@ struct Flower {
   // possible is helpful as anything reading them meanwhile (before we get to
   // their work item in the queue) will see the newer value, possibly avoiding
   // flowing an old value that would later be overwritten.
-#ifdef POSSIBLE_CONTENTS_INSERT_ORDERED
+  //
+  // This must be ordered to avoid nondeterminism. The problem is that our
+  // operations are imprecise and so the transitive property does not hold:
+  // (AvB)vC may differ from Av(BvC). Likewise (AvB)^C may differ from
+  // (A^C)v(B^C). An example of the latter is if a location is sent a null func
+  // and an i31, and the location can only contain funcref. If the null func
+  // arrives first, then later we'd merge null func + i31 which ends up as Many,
```

Comment:
i31 and `func` are on different type hierarchies so how could they possibly arrive at the same place?

It seems to me that the principled fix for this problem would be to fix our operations to have the algebraic properties we would expect, not to make the control flow deterministic. Is there a reason we can't or shouldn't do that?